### PR TITLE
Use metsrw for transfer METS creation

### PR DESCRIPTION
**Pull request by cole**
Not ready for review yet.

Connects to https://github.com/artefactual/archivematica/issues/998.
Connects to https://github.com/archivematica/Issues/issues/442.

### CODE REVIEW

File: src/MCPClient/lib/clientScripts/create_transfer_mets.py [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+                    file_obj.modificationtime.strftime("%Y-%m-%d"),
+                ),
+            ),
+        ),
+        ("original_name", escape(file_obj.originallocation)),
+    ) + get_premis_relationship_data(
+        file_obj.related_is_source_of, file_obj.related_has_source
+    )
+
+    return metsrw.plugins.premisrw.data_to_premis(premis_data)
```
Comment (sevein) [inline src/MCPClient/lib/clientScripts/create_transfer_mets.py]:
```suggestion
    return metsrw.plugins.premisrw.data_to_premis(
        premis_data, metsrw.plugins.premisrw.PREMIS_3_0_VERSION)
```

File: src/MCPClient/lib/clientScripts/create_transfer_mets.py [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+    Converts an File model object to a PREMIS event object via metsrw.
+
+    Returns:
+        metsrw.plugins.premisrw.premis.PREMISObject
+    """
+
+    premis_digest_algorithm = convert_to_premis_hash_function(file_obj.checksumtype)
+    premis_data = (
+        "object",
+        metsrw.plugins.premisrw.PREMIS_2_2_META,
```
Comment (sevein) [inline src/MCPClient/lib/clientScripts/create_transfer_mets.py]:
```suggestion
        metsrw.plugins.premisrw.PREMIS_3_0_META,
```

**Review by sevein — COMMENTED** [review]
I've added an example on how to use PREMIS 3 which we need (see [#24](https://github.com/archivematica/Issues/issues/24) for context).

File: src/MCPClient/lib/clientScripts/create_transfer_mets.py (hunk 2 above)
Comment (cole) [reply src/MCPClient/lib/clientScripts/create_transfer_mets.py]: Thanks! I was looking at the current behaviour of createprocessedstructmap. Will change.

File: src/MCPClient/lib/clientScripts/create_transfer_mets.py (hunk 1 above)
Comment (cole) [reply src/MCPClient/lib/clientScripts/create_transfer_mets.py]: Thanks! I was looking at the current behaviour of createprocessedstructmap. Will change.

**Review by sevein — COMMENTED** [review]
I've started using your branch for my rebuild work. I have a couple of comments:

1. The pipeline UUID is not being included in `metsHdr`. I only need this because when the transfer is sent to backlog, Storage Service [raises an error](https://github.com/artefactual/archivematica-storage-service/blob/4987a8c682d16ea38089acb7a9200a0c5f2200e5/storage_service/locations/models/package.py#L1784-L1790) when the METS doesn't include it. E.g.:
   ```xml
   <mets:metsHdr CREATEDATE="2015-12-16T22:38:48">
     <mets:agent OTHERTYPE="SOFTWARE" ROLE="CREATOR" TYPE="OTHER">
       <mets:name>39461beb-22eb-4942-88af-848cfc3462b2</mets:name>
       <mets:note>Archivematica dashboard UUID</mets:note>
     </mets:agent>
   </mets:metsHdr>
    ```

2. I've found that `Flocat` does not include the full relative path, e.g. I'd expect something like:
   ```xml
   <FLocat xlink:href="objects/V00154.MPG" LOCTYPE="OTHER" OTHERLOCTYPE="SYSTEM"/>
   ```

   But the XML that you're generating only reads `xlink:href="V00154.MPG"`. Not sure if the problem is here or in metsrw. When reading, I'd expect `fsentry.path` to be `objects/V00154.MPG` - at least that's how it's always been AFAIK. E.g. see https://github.com/artefactual/archivematica-storage-service/blob/4987a8c682d16ea38089acb7a9200a0c5f2200e5/storage_service/locations/models/package.py#L1799.

**Review by sevein — APPROVED** [review]
This is great!

I've been testing this and it's all working excepting the missing `metsHdr` agent details.

**Pull request merged**